kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -59,13 +56,23 @@ public class StructField {
     private final String name;
     private final DataType dataType;
     private final boolean nullable;
-    private final Map<String, String> metadata;
+    private final FieldMetadata metadata;
+
+    public StructField(
+        String name,
+        DataType dataType,
+        boolean nullable) {
+        this.name = name;
+        this.dataType = dataType;
+        this.nullable = nullable;
+        this.metadata = FieldMetadata.builder().build();
+    }
 
     public StructField(
         String name,
         DataType dataType,
         boolean nullable,
-        Map<String, String> metadata) {
+        FieldMetadata metadata) {
```

Comment:
nit: fix indentation while you're at it

Reply:
> Adds FieldMetadata class to store struct field metadata. We cannot use an ordinary Map here since field metadata stores values of different types.

Can this be done separately? In a refactor PR _before_ this PR?